alg/sha256_arm.c

```diff
@@ -98,9 +98,9 @@ SHA256_Transform_arm(uint32_t state[static restrict 8],
 	}
 
 	/* 4. Mix local working variables into global state. */
-	S0 = vaddq_u32(_state[0], S0);
-	S1 = vaddq_u32(_state[1], S1);
-	vst1q_u32(&state[0], S0);
-	vst1q_u32(&state[4], S1);
+	S[0] = vaddq_u32(_state[0], S[0]);
```

Comment:
And since we don't particularly care about `S` after this, I would `vst1q_u32(&state[0], vaddq_u32(_state[0], S[0]));` rather than storing the value in `S` and then copying it out.

Reply:
done